test/jdk/performance/client/RenderPerfTest/src/renderperf/RenderPerfTest.java

```diff
@@ -602,398 +799,1376 @@ private static BufferedImage makeManagedBI(final int type) {
+        }
+
+        protected void processTimes() {
+            nData = testTime.size();
+
+            if (!testTime.isEmpty()) {
+                // Ignore first 10% (warmup at the beginning):
+                final int thIdx = (int) Math.ceil(testTime.size() * 0.10);
+
+                final ArrayList<Long> times = new ArrayList<>(nData - thIdx);
+                for (int i = thIdx; i < nData; i++) {
+                    times.add(testTime.get(i));
+                }
+
+                // Sort values to get percentiles:
+                Collections.sort(times);
+                final int last = times.size() - 1;
+
+                if (USE_FPS) {
+                    scores[0] = fps(times.get(pctIndex(last, 0.5000))); //    50% (median)
+
+                    results[3] = fps(times.get(0)); // 0.0 (min)
+                    results[2] = fps(times.get(pctIndex(last, 0.1587))); // 15.87% (-1 stddev)
+                    results[1] = fps(times.get(pctIndex(last, 0.8413))); // 84.13% (+1 stddev)
+                    results[0] = fps(times.get(pctIndex(last, 1.0000))); // 100% (max)
+
+                    scores[1] = (results[2] - results[1]) / 2.0;
+                    scores[2] = millis(times.get(pctIndex(last, 0.5000))); //    50% (median)
+                } else {
+                    scores[0] = millis(times.get(pctIndex(last, 0.5000))); //    50% (median)
+
+                    results[0] = millis(times.get(0)); // 0.0 (min)
+                    results[1] = millis(times.get(pctIndex(last, 0.1587))); // 15.87% (-1 stddev)
+                    results[2] = millis(times.get(pctIndex(last, 0.8413))); // 84.13% (+1 stddev)
+                    results[3] = millis(times.get(pctIndex(last, 1.0000))); // 100% (max)
+
+                    scores[1] = (results[2] - results[1]) / 2.0;
+                    scores[2] = fps(times.get(pctIndex(last, 0.5000))); //    50% (median)
+                }
+            }
         }
 
-        private void report() {
-            if (skippedFrame > 0) {
-                System.err.println(skippedFrame + " frame(s) skipped");
+        protected String getResults() {
+            if (skippedFrames > 0) {
+                System.err.println(name + " : " + skippedFrames + " frame(s) skipped");
+            }
+            if (VERBOSE) {
+                return String.format("%-25s : %.3f (%.3f) %s [%.3f %s] (p00: %.3f p15: %.3f p50: %.3f p85: %.3f p100: %.3f %s) (%d frames)",
+                        name, scores[0], scores[1], (USE_FPS ? "FPS" : "ms"),
+                        scores[2], (USE_FPS ? "ms" : "FPS"),
+                        results[0], results[1], scores[0], results[2], results[3],
+                        (USE_FPS ? "FPS" : "ms"),
+                        nData);
             }
-            System.err.println(name + " : " + String.format("%.2f FPS", fps));
+            return String.format("%-25s : %.3f (%.3f) %s", name, scores[0], scores[1], (USE_FPS ? "FPS" : "ms"));
```

Comment:
Just a minor note:
It's not easy to understand the meaning of the value in brackets from the output. Some hints like a table header with column names printed before the results could be useful here.

java -jar dist/RenderPerfTest.jar
Using GraphicsConfiguration(s): [0:0][MTLGraphicsConfig[Display 1]] 
ArgbSurfaceBlitImage[1]   : 60.497 (10.408) FPS
ArgbSwBlitImage[1]        : 11.226 (0.569) FPS
BgrSurfaceBlitImage[1]    : 59.700 (10.446) FPS

Reply:
I will add table header and also refine statistics to give MAD (Median absolute deviation) to give a robust estimator for result dispersion.